helm/amazon-ec2-metadata-mock/templates/_helpers.tpl

```diff
@@ -54,3 +62,41 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Get the default node selector term prefix.
+
+In 1.14 "beta.kubernetes.io" was deprecated and is scheduled for removal in 1.18.
+See https://v1-14.docs.kubernetes.io/docs/setup/release/notes/#deprecations
+*/}}
+{{- define "amazon-ec2-metadata-mock.defaultNodeSelectorTermsPrefix" -}}
+    {{- semverCompare "<1.14" .Capabilities.KubeVersion.Version | ternary "beta.kubernetes.io" "kubernetes.io" -}}
```

Comment:
1. It turns out that `.Version` is only present in Helm 3.  Here is a [workaround](https://github.com/aws/aws-node-termination-handler/commit/e16065c826fc2fc350e15476ffca34f9222431e8#diff-d4534d4837f947132873af8b9dc3c6a9L74) used to support Helm 2.
2. An issue was encountered when the version includes [build metadata](https://semver.org/#spec-item-10).  Here is a [workaround](https://github.com/aws/aws-node-termination-handler/commit/43a69e0f41f2d7ed97e7199f5026dfc2c7eec6e4#diff-d4534d4837f947132873af8b9dc3c6a9R74).

Reply:
Also, might want to do <1.19 since beta is supported on all GA released k8s versions. We recently found out that this is an immutable field in k8s and will not cleanly perform a helm upgrade if it changes.  It's not a big deal here since the use-case is not to constantly upgrade.